api/v1alpha1/api.go

```diff
@@ -80,6 +80,101 @@ type AIGatewayRouteSpec struct {
 	// Currently, the filter is only implemented as an external process filter, which might be
 	// extended to other types of filters in the future. See https://github.com/envoyproxy/ai-gateway/issues/90
 	FilterConfig *AIGatewayFilterConfig `json:"filterConfig,omitempty"`
+
+	// LLMRequestCosts specifies how to capture the cost of the LLM-related request, notably the token usage.
+	// The AI Gateway filter will capture each specified number and store it in the Envoy's dynamic
+	// metadata per HTTP request. The namespaced key is "io.envoy.ai_gateway",
+	//
+	// For example, let's say we have the following LLMRequestCosts configuration:
```

Comment:
@arkodg appreciate it if you could double check this doc comment

Reply:
looks good, do you want to highlight `Distinct` match so individual limits are applies to unique users - `one`, `two` .... defined in `x-user-id` ?

Reply:
good catch! fixed in https://github.com/envoyproxy/ai-gateway/pull/103/commits/e6370a88cbf651ef63fec1981f22adeaf5b1f3b8